src/transformers/training_args.py

```diff
@@ -324,18 +324,15 @@ class TrainingArguments:
     def __post_init__(self):
         if self.disable_tqdm is None:
             self.disable_tqdm = logger.getEffectiveLevel() > logging.WARN
-        if self.evaluate_during_training is not None:
-            self.evaluation_strategy = (
-                EvaluationStrategy.STEPS if self.evaluate_during_training else EvaluationStrategy.NO
-            )
+        if self.evaluate_during_training is True:
+            self.evaluation_strategy = EvaluationStrategy.STEPS
```

Comment:
Had overlooked this before, but this overwrites `self.evaluation_strategy`, right?  Wouldn't it be better to check if `self.evaluation_strategy` was set to different value than `NO` before setting it?

Reply:
There is the deprecation warning already. I think it should overwrite the `self.evaluation_strategy` since it's the old argument versus the new. People are not supposed to use both at the same time.
Can add that to the warning message if you think it's best.

Reply:
Alright, works for me!